src/figuresClasses.ts

```diff
@@ -1,19 +1,91 @@
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
+}
+
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
+}
+
 export interface Figure {
+  color: Color;
+  shape: Shape;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape = Shape.triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('side length cannot be negative or zero');
+    }
 
+    const longestSide = Math.max(a, b, c);
+    const otherSidesSum = a + b + c - longestSide;
+
+    if (longestSide >= otherSidesSum) {
+      // eslint-disable-next-line max-len
+      throw new Error('longest side cannot be equal to or greater than the sum of two other sides');
+    }
+  }
+
+  getArea(): number {
+    const { a, b, c } = this;
+    const semPer = 0.5 * (a + b + c);
+    const area = Math.sqrt(semPer * (semPer - a) * (semPer - b) * (semPer - c));
+
+    return Math.floor(area * 100) / 100;
```

Comment:
Remember about DRY. You are using this part of code at least twice, create function for this `Math.floor(... * 100) / 100`